### ISSUE TITLE
Issue with find: doesn't flow

### ISSUE THREAD

**Issue by whitecolor** [donnut/typescript-ramda#88]
```js
type Task = {id: number}
let tasks: Task[] = []
R.find(task => task.id === 1, tasks) // this works

R.find(task => task.id === 1)(tasks) // this doesn't work works
```

Is it possible to make it flow in second case?

**Comment by whitecolor** [comment]
The same problem actually with other methods:

```js
let obj: { id: number } = { id: 1 }
R.prop<string>('id')(obj)
```

And in general I believe that is some limitation of TS. In this cases it is more appropriate to use not ramda but simple mapping function like `(_) => _.id` that will flow perfectly.

What do you think?

**Comment by donnut** [comment]
I changed the definition of the curried find to:
```find<T>(fn: (a: T) => boolean): <T>(list: T[]) => T;```
and now it works as expected, see pr #91

**Comment by donnut** [comment]
Concerning the 'prop'-issue. This is a know limitiation of TS. Version 1.8+ should fix that, see #10

**Comment by whitecolor** [comment]
@donnut 
> Concerning the 'prop'-issue. This is a know limitiation of TS. Version 1.8+ should fix that, see #10

I'm using 2.1-dev. What signature are you talking about? It seem that string based typing is not supported.
